### PR TITLE
Cpfcnpjupdate2

### PR DESCRIPTION
**Pull request by noodlemoodle**
- Add FormattedString sub-type of String
- Override javasetter for FormattedString to strip the data of formatting
- Override tablecellformatter for FormattedString to apply format before adding the data
- Update CPF and CNPJ fields: https://github.com/nanoPayinc/NANOPAY/pull/12543

### CODE REVIEW

File: src/foam/java/refinements.js [hunk 1]
```diff
@@ -1612,6 +1612,30 @@ foam.CLASS({
...
+    TODO: Add the ability to get a formatted version of the data 
+  `,  
+
+  properties: [
+    {
+      name: 'javaSetter',
+      factory: function() {
+        return `
+          assertNotFrozen();
+          val = val.replaceAll("[^\\\\\d]", "");
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:1631]: Add a comment to explain what the regex matches.

File: src/foam/core/types.js [hunk 2]
```diff
@@ -106,6 +106,20 @@ foam.CLASS({
...
 
+foam.CLASS({
+  package: 'foam.core',
+  name: 'FormattedString',
+  extends: 'String',
+  documentation: 'A delimiter separated string of digits',
+
+  properties: [
+    {
+      name:'formatter',
```
Comment (kgrgreer) [inline src/foam/core/types.js:117]: Document what format this value should be supplied as.

**Comment by noodlemoodle** [comment]
addressed comments https://github.com/kgrgreer/foam3/pull/188/commits/eed79631ec64df91ba9ac1cb404d23912b41062a @kgrgreer

File: src/foam/core/types.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/core/types.js:117]: @noodlemoodle perhaps an example.

**Pull request merged**